src/components/ViewList.js

```diff
@@ -1,68 +1,56 @@
-import React, { useContext, useState, useEffect } from 'react';
+import React, { useContext } from 'react';
 import { ListContext } from '../context/ListContext';
 import { Link } from 'react-router-dom';
+import calculateEstimate from '../lib/estimates';
 
 const ViewList = () => {
-
   // If the list is empty, add a prompt and link to Add Items
   let currentList = useContext(ListContext);
-  let token = currentList.token;
-
-  const [itemsPurchased, setItemsPurchased] = useState({});
 
   const handleCheck = async (e) => {
-    e.persist();
-    const item = e.target.name;
-    const isChecked = e.target.checked;
-    await setItemsPurchased((prevItemsPurchased) => ({
-      ...prevItemsPurchased,
-      [item]: isChecked,
-    }));
-    updateDatabase(e);
-  };
+    const timeNow = new Date().getTime() / 1000;
+    // retrieve event informationt
+    let docId = e.target.id;
+    let currentRef = currentList.itemsRef.doc(docId);
 
-  useEffect(() => {
-    const handleTiming = () => {
-      let currentTime = Date.now();
-      console.log(currentTime);
+    //find context for current item
+    const currentItem = currentList.userList.find((item) => {
+      return item.id == docId;
+    });
 
-      const lastPurchasedTimeArray =
-        currentList.userList &&
-        currentList.userList.map((element) => {
-          const container = {};
-          container.itemName = element.itemName;
-          container.t = element.lastPurchased;
-          return container;
-        });
+    // store current item data
+    let numberOfPurchases = 1;
+    currentItem.numberOfPurchases
+      ? (numberOfPurchases = currentItem.numberOfPurchases + 1)
+      : (numberOfPurchases = 1);
+    let lastEstimate =
+      parseInt(currentItem.lastEstimate) || parseInt(currentItem.timeFrame);
```

Comment:
It looks like the timeFrame was removed. Is this trying to be backwards compatible? 
If so, the usual practice is to run a script on the database to rename everything, but since this is a demo project, you could ask the team if it is ok to clean up the database and start from a clean slate that will only have lastEstimate, otherwise this may look confusing to you going forward because you can't see a timeFrame anywhere in the code.